webpack.config.js

```diff
@@ -3,116 +3,246 @@
-	);
+/**
+ * Collect information about all packages
+ * and their entry files using entry-files.json files.
+ *
+ * @return {Object[]} Array of items with information
+ */
+function getEntryFiles() {
+	const entries = [];
 
 	// Only include directories that contains a entry-files.json file.
-	dirs = dirs.filter( ( dir ) =>
-		fs.existsSync( path.resolve( dir, 'entry-files.json' ) )
+	const files = globSync(
+		'./packages/(mu-plugins|plugins|themes)/**/entry-files.json',
+		{ onlyFiles: true }
 	);
 
-	dirs.forEach( ( dir ) => {
-		const entryFiles = require( `${ dir }/entry-files.json` ); // eslint-disable-line
+	files.forEach((file) => {
+		const entryFiles = require(file); // eslint-disable-line
 
-		entryFiles.forEach( ( entry ) => {
-			/**
-			 * MiniCSSExtractPlugin handles css files. Rename them and ignore in
-			 * IgnoreEmitPlugin to allow as direct entry files.
-			 */
-			entries[
-				`${ dir }|${ entry.replace( '.css', '.css-entry-file' ) }`
-			] = `${ dir }/src/${ entry }`;
-		} );
-	} );
+		entries.push({
+			dir: path.dirname(file),
+			files: entryFiles.map((entryFile) => `src/${entryFile}`),
+		});
+	});
 
 	return entries;
 }
 
 /**
- * Config
+ * Closure function to pretty print information about packages and
+ * their entry files of assets.
  */
-const config = {
-	...defaultConfig,
-	entry: async () => getEntries(),
-	output: {
-		path: process.cwd(),
-		filename: ( { chunk } ) => getBuildPath( chunk.name ),
-	},
-	optimization: {
-		...defaultConfig.optimization,
-		splitChunks: {
-			cacheGroups: {
-				style: {
-					type: 'css/mini-extract',
-					test: /[\\/]style(\.module)?\.(sc|sa|c)ss$/,
-					chunks: 'all',
-					enforce: true,
-					name( _, chunks ) {
-						return getBuildPath(
-							chunks[ 0 ].name.replace(
-								'.css-entry-file',
-								'.css'
-							)
-						);
+const packagesTable = (() => {
+	const entriesForTable = {};
+
+	const p = new Table({
+		columns: [
+			{ name: 'dir', title: 'Package', alignment: 'left' },
+			{ name: 'files', title: 'Entry Files', alignment: 'left' },
+		],
+	});
+
+	return {
+		printTable: () => {
+			const dirs = Object.keys(entriesForTable);
+			dirs.forEach((dir, i) => {
+				const entryForTable = entriesForTable[dir];
+				Object.values(entryForTable).forEach((entryForTableItem, j) => {
+					p.addRow({
+						dir: j === 0 ? dir : '',
+						files: entryForTableItem.join(' + '),
+					});
+				});
+
+				if (dirs.length - 1 > i) {
+					p.addRow({
+						dir: '───────────────────────────────────────────────────',
+						files: '────────────────────────────────',
+					});
+				}
+			});
+
+			p.printTable();
+		},
+
+		logEntry: (dir, file) => {
+			const fileName = path.parse(file).name;
+
+			if (typeof entriesForTable[dir] === 'undefined') {
+				entriesForTable[dir] = [];
+			}
+
+			if (typeof entriesForTable[dir][fileName] === 'undefined') {
+				entriesForTable[dir][fileName] = [];
+			}
+			entriesForTable[dir][fileName].push(file);
+		},
+	};
+})();
```

Comment:
This is nice 👍 , but is it something we need to add in project-base?

Next time I would like stuff like this to be in a seperate PR/issue, where a discussion about a spesific functionallity can take place. It makes reviewing easier when I can focus on what the PR tries to solve.